parser/src/parser/grammar.lalrpop

```diff
@@ -283,16 +285,32 @@ IntegrityStmts: Vec<IntegrityStmt> = {
             });
         }
         Ok(integrity_stmts)
-    }
+    },
+    
 }
 
-IntegrityStmt: IntegrityStmt = {
+IntegrityStmtGroup: Vec<IntegrityStmt> = {
     "let" <name: Identifier> "=" <integrity_variable_type: IntegrityVariableType> =>
-        IntegrityStmt::Variable(Variable::new(name, integrity_variable_type)),
-    "enf" <lhs: IntegrityExpr> "=" <rhs: IntegrityExpr> =>
+        vec![IntegrityStmt::Variable(Variable::new(name, integrity_variable_type))],
+    "enf" <integrity_constraint: IntegrityConstraintExpr> => vec![integrity_constraint],
+    "match" "enf" ":" <integrity_stmts: IntegrityConstraintExpr+> => integrity_stmts
+}
+
+IntegrityConstraintExpr: IntegrityStmt = {
+    <lhs: IntegrityExpr> "=" <rhs: IntegrityExpr> =>
         IntegrityStmt::Constraint(ConstraintType::Inline(IntegrityConstraint::new(lhs, rhs)), None),
-    "enf" <evaluator_fn_call: EvaluatorFunctionCall> =>
+    <evaluator_fn_call: EvaluatorFunctionCall> =>
         IntegrityStmt::Constraint(ConstraintType::Evaluator(evaluator_fn_call), None),
+    <integrity_stmt_with_selector: IntegrityStmtWithSelector> =>
+        integrity_stmt_with_selector,
+}
+
+IntegrityStmtWithSelector: IntegrityStmt = {
```

Comment:
nit: This could probably be `IntegrityConstraintWithSelector`, since there are no variables allowed here